flaky `test: test_canica.py::test_percentile_range`

It seems to fail "once in a while"

https://github.com/nilearn/nilearn/actions/runs/4301638495/jobs/7499114761#step:8:16363
https://github.com/nilearn/nilearn/actions/runs/4292758055/jobs/7479594709#step:8:16252
https://github.com/nilearn/nilearn/actions/runs/4277980448/jobs/7447269258#step:8:16124
https://github.com/nilearn/nilearn/actions/runs/4251899059/jobs/7394830022#step:8:16113

```bash
FAILED nilearn/decomposition/tests/test_canica.py::test_percentile_range - assert 2 == 1
 +  where 2 = sum([False, False, False, False, False, False, ...])
```

--- a/nilearn/decomposition/tests/test_canica.py
+++ b/nilearn/decomposition/tests/test_canica.py
@@ -169,16 +169,14 @@ def test_percentile_range():
     canica = CanICA(n_components=edge_case, threshold=float(edge_case))
     with warnings.catch_warnings(record=True) as warning:
         canica.fit(data)
-        # Filter out deprecation warnings
-        not_deprecation_warning = [
-            not issubclass(w.category, (DeprecationWarning, FutureWarning))
+        # ensure a single warning is raised
+        # filter out deprecation warnings
+        warning_messages = [
+            "obtained a critical threshold" in str(w.message)
             for w in warning
+            if not issubclass(w.category, (DeprecationWarning, FutureWarning))
         ]
-        assert sum(not_deprecation_warning) == 1  # ensure single warning
-        idx_critical_warning = not_deprecation_warning.index(True)
-        assert "critical threshold" in str(
-            warning[idx_critical_warning].message
-        )
+        assert sum(warning_messages) == 1
 
 
 def test_masker_attributes_with_fit():

